wrappers/python/simtk/openmm/app/forcefield.py

```diff
@@ -263,17 +263,17 @@ def loadFile(self, files):
                     template = ForceField._TemplateData(resName)
                     if 'override' in residue.attrib:
                         template.overrideLevel = int(residue.attrib['override'])
-                    atomIndices = {}
-                    for atom in residue.findall('Atom'):
+                    atomIndices = template.atomIndices
+                    for ia, atom in enumerate(residue.findall('Atom')):
                         params = {}
-                        for key in atom.attrib:
-                            if key not in ('name', 'type'):
-                                params[key] = _convertParameterToNumber(atom.attrib[key])
-                        atomName = atom.attrib['name']
+                        a_attrib = atom.attrib
+                        atomName = a_attrib.pop('name')
```

Comment:
This line makes me a little nervous.  We load the XML into the etree data structure, then copy information from it into our own data structures.  But here you're directly modifying the etree data structures.  That's the sort of thing that could easily lead to bugs later on, for example if someone writes more code that looks at those structures and expects them to still reflect the content of the file.

What about keeping the parsing code as it was before and just adding the single line `template.atomIndices = atomIndices` at the end?